client/lib/form-fields/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { union } from 'lodash';
+
+export const formFields = WrappedComponent => {
```

Comment:
For naming of higher-order components, I typically see either as verb (`connect`, `localize`) or using some sort of preposition (recompose's `withState`, React Router's `withRouter`). For consistency's sake, should we name this something like `trackForm` or `withFormTracking` ?